baybe/utils/chemistry.py

```diff
@@ -72,158 +71,120 @@ def name_to_smiles(name: str) -> str:
 
 @lru_cache(maxsize=None)
 @_disk_cache
-def _smiles_to_mordred_features(smiles: str) -> np.ndarray:
-    """Memory- and disk-cached computation of Mordred descriptors.
+def _molecule_to_fingerprint_features(
+    fingerprint_encoder: BaseFingerprintTransformer,
+    molecule: str | Chem.PropertyMol.PropertyMol,
+) -> np.ndarray:
```

Comment:
It is a bit weird that the function for calculating a single fingerprint returns an `np.ndarray` while the one returning a list of them returns a `pd.dataframe`. Can this yield problems/confusion or is this function only intended to be used internally anyway?

Reply:
internal helper for the one without "_", 
for the other we need df to have named indices, here where it is only 1 row it is not needed as anyway this gets processed by the other func when concat

Reply:
All good here, the function has now been discussed in various other threads and I've also fixed the remaining issues like input type and docstring. So will close this thread. You can have another look and reopen if needed